### ISSUE TITLE
unique symbol incorrectly widens to symbol when assigning to a const or readonly property

### ISSUE THREAD

**Issue by kpdonn** [Microsoft/TypeScript#23263]
**Search Terms:** 
unique symbol, literal, widen, reassign, const, readonly

**Code**

```ts
const t1 = Symbol("t1")
const t1Copy = t1
const result1: typeof t1 = t1 // no error
const result2: typeof t1 = t1Copy // error: type symbol not assignable to unique symbol

class T2 {
    static readonly sym = Symbol("T2")
    readonly kind = T2.sym
}

const inst = new T2()
const result3: typeof T2.sym = T2.sym // no error
const result4: typeof T2.sym = inst.kind // error: type symbol not assignable to unique symbol

declare function genericIdentity<T>(arg: T): T
const unexpectedError: typeof t1 = genericIdentity(t1)

declare function genericSymbolNested<T extends symbol>(arg: T): { prop: T }
const unexpectedError2: { prop: typeof t1 } = genericSymbolNested(t1)
```

**Expected behavior:**
No errors. `t1Copy` and `inst.kind` should retain their respective literal `unique symbol` types the same way string literals would because they are declared with `const` and `readonly`.

**Actual behavior:**
Errors on `result2` and `result4` because `t1Copy` and `inst.kind` are widened to `symbol`. When assigning a `unique symbol` variable to a new variable the type is always widened to `symbol` unless an explicit type annotation is added.

**Playground Link:** 
[Link](https://www.typescriptlang.org/play/#src=const%20t1%20%3D%20Symbol(%22t1%22)%0D%0Aconst%20t1Copy%20%3D%20t1%0D%0Aconst%20result1%3A%20typeof%20t1%20%3D%20t1%20%2F%2F%20no%20error%0D%0Aconst%20result2%3A%20typeof%20t1%20%3D%20t1Copy%20%2F%2F%20error%3A%20type%20symbol%20not%20assignable%20to%20unique%20symbol%0D%0A%0D%0Aclass%20T2%20%7B%0D%0A%20%20%20%20static%20readonly%20sym%20%3D%20Symbol(%22T2%22)%0D%0A%20%20%20%20readonly%20kind%20%3D%20T2.sym%0D%0A%7D%0D%0A%0D%0Aconst%20inst%20%3D%20new%20T2()%0D%0Aconst%20result3%3A%20typeof%20T2.sym%20%3D%20T2.sym%20%2F%2F%20no%20error%0D%0Aconst%20result4%3A%20typeof%20T2.sym%20%3D%20inst.kind%20%2F%2F%20error%3A%20type%20symbol%20not%20assignable%20to%20unique%20symbol%0D%0A%0D%0Adeclare%20function%20genericIdentity%3CT%3E(arg%3A%20T)%3A%20T%0D%0Aconst%20unexpectedError%3A%20typeof%20t1%20%3D%20genericIdentity(t1)%0D%0A%0D%0Adeclare%20function%20genericSymbolNested%3CT%20extends%20symbol%3E(arg%3A%20T)%3A%20%7B%20prop%3A%20T%20%7D%0D%0Aconst%20unexpectedError2%3A%20%7B%20prop%3A%20typeof%20t1%20%7D%20%3D%20genericSymbolNested(t1)%0D%0A)

**Related Issues:**
Did not notice any similar bug reports, but the design notes from #17501 seem to confirm that the current behavior is a bug:

> - Like literal types, these types contain "freshness" which can be widened away to produce a "regular" type.
> - When you alias a symbolic const with another const, it has the type `typeof [[that original variable]]`.

Which implies they should work like other literal types which would not be widened in the example above.

**Comment by kpdonn** [comment]
Additional cases with generics:

```ts
declare const sym: unique symbol

declare function genericIdentity<T>(arg: T): T
const unexpectedError: typeof sym = genericIdentity(sym)

declare function genericSymbolNested<T extends symbol>(arg: T): { prop: T }
const unexpectedError2: { prop: typeof sym } = genericSymbolNested(sym)

// Compare with the following equivalent code using literal strings:

declare const literalString: "literal"
const noErrorWithString: typeof literalString = genericIdentity(literalString)

declare function genericStringNested<T extends string>(arg: T): { prop: T }
const noErrorWithString2: { prop: typeof literalString } = genericStringNested(literalString)
```
[Playground link](https://www.typescriptlang.org/play/#src=declare%20const%20sym%3A%20unique%20symbol%0D%0A%0D%0Adeclare%20function%20genericIdentity%3CT%3E(arg%3A%20T)%3A%20T%0D%0Aconst%20unexpectedError%3A%20typeof%20sym%20%3D%20genericIdentity(sym)%0D%0A%0D%0Adeclare%20function%20genericSymbolNested%3CT%20extends%20symbol%3E(arg%3A%20T)%3A%20%7B%20prop%3A%20T%20%7D%0D%0Aconst%20unexpectedError2%3A%20%7B%20prop%3A%20typeof%20sym%20%7D%20%3D%20genericSymbolNested(sym)%0D%0A%0D%0A%2F%2F%20Compare%20with%20the%20following%20equivalent%20code%20using%20literal%20strings%3A%0D%0A%0D%0Adeclare%20const%20literalString%3A%20%22literal%22%0D%0Aconst%20noErrorWithString%3A%20typeof%20literalString%20%3D%20genericIdentity(literalString)%0D%0A%0D%0Adeclare%20function%20genericStringNested%3CT%20extends%20string%3E(arg%3A%20T)%3A%20%7B%20prop%3A%20T%20%7D%0D%0Aconst%20noErrorWithString2%3A%20%7B%20prop%3A%20typeof%20literalString%20%7D%20%3D%20genericStringNested(literalString)%0D%0A)

**Comment by mhegazy** [comment]
Type of a unique symbol is widened unless it is of the form `const sym = Symbol()`. the intent here is not clear. this was also done to avoid breaking changes.

**Comment by falsandtru** [comment]
You don't have any reason to widen the type here. This change never breaks any valid code. Then this is pure improvement. You don't have any reason to avoid this improvement.

**Comment by kpdonn** [comment]
> the intent here is not clear.

The intent is that unique symbols should work the same way as string literals. They are conceptually the exact same thing: a type that has exactly one possible value. The use cases for the behavior I described are the same as the use cases for string literals. 

I agree with @falsandtru that I don’t understand what cases you are concerned about where this would be a negative breaking change. 

Also I just want to say that it’s very frustrating to go through the effort of putting together a very clear, minimally reproducible bug report only to have it apparently glanced at for a few seconds and assumed invalid. When I ran into this issue it wasn’t remotely clear what the root cause was. It took a decent amount of effort to narrow it down to the very straightforward example I posted here. It also took time to do the research to try to confirm my feeling that this was an unintentional bug, which the design notes from #17501 did seem to explicitly confirm.

I realize you have many issues to go through and that most of them are invalid or duplicates which is why I try as much as possible to ensure that my issues are both valid and not duplicates. I try to respect your time but frankly I feel that respect is not always returned.
